### ISSUE TITLE
Upgrade pinned version of pip3

### ISSUE THREAD

**Issue by lilatomic** [GeoNode/geoserver-docker#84]
pip3 is currently pinned to version 9.0.3 [here](https://github.com/GeoNode/geoserver-docker/blob/ba1eb5eb9708bd93d43a268f03012da4805ad684/Dockerfile#L82), and has been for about 4 years. We should probably unpin it, or at least upgrade it.
I'm happy to submit a patch for whichever we prefer.

**Comment by t-book** [comment]
I do agree.

There has been a problem back in the days: https://github.com/GeoNode/geoserver-docker/commit/21b14c2ca15b53c882759729833f172b87954278 with an upgraded version.
before unpinning it we should test this carefully.

**Comment by frafra** [comment]
We could implement GitHub actions to run a docker build and check for errors in advance.

**Comment by lilatomic** [comment]
Right. Seems like that was a fix for #20 , where the problem was pip 10.0.0 . Seems that pip 10.0.0 changed its internals; but, since it was installed by apt but then upgraded by us, that posed problems (pip warns you about doing this, although this is the first time I've seen that actually causing issues).

I tried just removing the line to upgrade pip and use the version provided by the package manager [20.3.4](https://pip.pypa.io/en/stable/news/#v20-3-4). It looks like everything installed fine. In general the one installed with the package manager is a bit old (about 1.3 years) but regarded as stable.

What sorts of acceptance tests should I run to qualify the image?

Using GitHub actions to verify that it still builds is a neat idea.

**Comment by frafra** [comment]
1. Build test
2. Run the image and check if the service is reachable over HTTP within a minute

Those two very simple tests that would catch the biggest regressions, in my opinion.

**Comment by giohappy** [comment]
This repo will be aligned and upgraded soon https://github.com/GeoNode/geoserver-docker/issues/87
